src/sage/algebras/letterplace/free_algebra_letterplace.pyx

```diff
@@ -624,15 +636,17 @@ cdef class FreeAlgebra_letterplace(Algebra):
         cdef list out = []
         cdef list tmp
         cdef list names = self.latex_variable_names()
-        for i from 0<=i<nblocks:
+        cdef int N = len(self._names)
+        for i in range(nblocks):
             base = i*ngens
             tmp = [(j, E[base+j]) for j in range(ngens) if E[base+j]]
             if not tmp:
                 continue
             var_ind, exp = tmp[0]
             if len(tmp) > 1 or exp > 1:
                 raise NotImplementedError("\n  Apparently you tried to view the letterplace algebra with\n  shift-multiplication as the free algebra over a finitely\n  generated free abelian monoid.\n  In principle, this is correct, but it is not implemented, yet.")
-
+            if self._nb_slackvars and var_ind == N:
+                continue
```

Comment:
Same question here. Is this fixing a bug? Implementing a case currently not implemented?
Example / test case please